tests/foreman/ui_airgun/test_contentview.py

```diff
@@ -1571,6 +1569,210 @@ def test_positive_remove_cv_version_from_default_env(session, module_org):
+        assert puppet_module_name in cvv['puppet_modules']['table'][0]['Name']
+        assert yum_repo_name == cvv['yum_repositories']['table'][0]['Name']
+        cvv = session.contentview.search_version(cv['name'], VERSION)[0]
+        assert qe_lce.name in cvv['Environments']
+        # remove the content view version from QE Environment
+        session.contentview.remove_version(cv['name'], VERSION, False, [qe_lce.name])
+        cvv = session.contentview.search_version(cv['name'], VERSION)[0]
+        assert qe_lce.name not in cvv['Environments']
+        # promote again to QE
+        result = session.contentview.promote(cv['name'], VERSION, qe_lce.name)
+        assert 'Promoted to {}'.format(qe_lce.name) in result['Status']
+        cvv = session.contentview.read_version(cv['name'], VERSION)
+        assert puppet_module_name in cvv['puppet_modules']['table'][0]['Name']
+        assert yum_repo_name == cvv['yum_repositories']['table'][0]['Name']
+        cvv = session.contentview.search_version(cv['name'], VERSION)[0]
+        assert qe_lce.name in cvv['Environments']
+
+
+@upgrade
+@tier2
+def test_positive_delete_cv_promoted_to_multi_env(session, module_org):
+    """Delete published content view with version promoted to multiple
+     environments
+
+    :id: f16f2db5-7f5b-4ebb-863e-6c18ff745ce4
+
+    :Steps:
+
+        1. Create a content view
+        2. Add a yum repo and a puppet module to the content view
+        3. Publish the content view
+        4. Promote the content view to multiple environment Library -> DEV
+            -> QE -> STAGE -> PROD
+        5. Delete the content view, this should delete the content with all
+            it's published/promoted versions from all environments
+
+    :expectedresults: The content view doesn't exists
+
+    :CaseLevel: Integration
+    """
+    repo = RepositoryCollection(
+        repositories=[YumRepository(url=FAKE_0_YUM_REPO)])
+    repo.setup(module_org.id)
+    cv, lce = repo.setup_content_view(module_org.id)
+    repo_name = repo.repos_info[0]['name']
+    with session:
+        cvv = session.contentview.read_version(cv['name'], VERSION)
+        assert repo_name == cvv['yum_repositories']['table'][0]['Name']
+        cvv = session.contentview.search_version(cv['name'], VERSION)[0]
+        assert lce['name'] in cvv['Environments']
+        lce_values = session.lifecycleenvironment.read(lce['name'])
+        assert len(lce_values['content_views']['resources']) == 1
+        assert lce_values['content_views']['resources'][0]['Name'] == cv['name']
+        session.contentview.remove_version(
+            cv['name'], VERSION, False, [ENVIRONMENT, lce['name']])
+        cvv = session.contentview.search_version(cv['name'], VERSION)[0]
+        assert lce['name'] not in cvv['Environments']
+        session.contentview.delete(cv['name'])
+        lce_values = session.lifecycleenvironment.read(lce['name'])
+        assert len(lce_values['content_views']['resources']) == 0
```

Comment:
Think we are missing many envs: implementaed only Library-> QE

```
Promote the content view to multiple environment Library -> DEV -> QE -> STAGE -> PROD

```

Reply:
that is redundant as we need to verify that content is actually removed, not promotion to different envs. Anyway promote to two different environments. Will change the docstring. Thanks

Reply:
Basically that test case has almost zero sense as we cannot remove cv before disassociate it from promoted environments, so at least I add verification for lce side, so we have at least some positive coverage